### DISCUSSION TITLE
sim.plot3D

### DISCUSSION THREAD

**Discussion by wllshr123** [NanoComp/meep#2429] [Q&A]
I try to add a plane source of XY plane in a sim space (2, 5, 10)，the simple test code diagram is as follows（Add a source on the basis of [this example](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#visualizing-3d-structures)）：
```
import meep as mp
import numpy as np
import math

cell_size = mp.Vector3(2,5,10)
wvl = 1.05  # wavelength
fcen = 1/wvl

vertices = [mp.Vector3(-1,0),
            mp.Vector3(-0.5,math.sqrt(3)/2),
            mp.Vector3(0.5,math.sqrt(3)/2),
            mp.Vector3(1,0),
            mp.Vector3(0.5,-math.sqrt(3)/2),
            mp.Vector3(-0.5,-math.sqrt(3)/2)]

src_cmpt = mp.Ez
src_pt = mp.Vector3(0,0,-1)
sources = [mp.Source(src=mp.GaussianSource(fcen,fwidth=0.1*fcen),
                     size=mp.Vector3(2,5,0),
                     center=src_pt,
                     component=src_cmpt)]

geometry = [mp.Prism(vertices, height=1.0, material=mp.Medium(index=3.5)),
            mp.Cone(radius=1.0, radius2=0.1, height=2.0, material=mp.air)]

sim = mp.Simulation(resolution=50,
                    cell_size=cell_size,
                    sources=sources,
                    )

sim.plot3D()
```
![image](https://user-images.githubusercontent.com/124772453/223649259-a6984df3-1656-4ba8-9598-c50a619b495d.png)

Why is the source plane drawn XZ plane?

**Comment by smartalecH** [comment]
(cc @SkandanC who recently overhauled this feature)

**Comment by jianlongzaitian** [comment]
I met the same problem? Did you handle it?

**Comment by gummyequation** [comment]
I have run into the same problem with a much busier settup, as of now it is still unresolved with no replies. I will link my issue/discussion
https://github.com/NanoComp/meep/discussions/2861
![image](https://github.com/user-attachments/assets/913494df-2080-4f6b-a258-db250a1b1b8f)

**Comment by jianlongzaitian** [comment]
Luckly, after my check, the issues only occurs in the commend "plot3D()", i runned the "Meep Project #5 — Thermal Radiation Spectra of Plasmonic Metamaterials"(http://www.simpetus.com/projects.html) and get a wrong 3D plot and a almost right result. Therefore, I think we can temporarily ignore the 3D model if we need meep for simulation urgently.
![fig_plot3D](https://github.com/user-attachments/assets/2294fa21-ea7b-4cda-bfd3-e31a446dd183)
![Figure_1](https://github.com/user-attachments/assets/bf73bbbc-4083-47c7-a415-1a779b5b6d6e)

**Reply by jianlongzaitian** [reply]
And, if you want get the right 3D model, just change the set "(sx, sy, 0)" to "(sx, 0, sy)", however, i think you can't get the right result when you run in this model.
